### DISCUSSION TITLE
iOS - Push Notification with custom data not being received in RemoteMessage

### DISCUSSION THREAD

**Discussion by ibnukipa** [invertase/react-native-firebase#5880] [Q&A] [answered]
I have correct setup the iOS project and why the remoteMessage is corrupted?

<img width="1013" alt="Screen Shot 2021-11-25 at 21 50 22" src="https://user-images.githubusercontent.com/12580174/143462360-87008c9d-23e6-4ada-9104-c2288b41f14b.png">
.

**Comment by mikehardy** [comment, marked as answer]
Not nearly enough information to understand the problem, nor propose any guesses.
My guess is that what you think you are sending as a remote message, is not what you are actually sending.

**Reply by ibnukipa** [reply]
@mikehardy Sorry I'm not really good to asking. So, the `RemoteMessage.data` is always empty object on iOS but not in Android. is there any step I missed?

**Reply by mikehardy** [reply]
With the information you  have provided, I can not say anything for certain.
All I can say is that when I my handlers are called in response to FCM, I have contents in the data property of the FCM JSON.

**Reply by ibnukipa** [reply]
yeah, that's why I'm asking. with the same FCM JSON that send from the firebase admin, in my Android app is completely received.

in Android, `messaging().onNotificationOpenedApp()` or `messaging().getInitialNotification()` returning this
```
RemoteMessage {
 data: { screen: "profile", id: "123"  },
 notification: { body: 'new msg', title: 'Incoming Message' }
}
```

but in iOS, `messaging().onNotificationOpenedApp()` or `messaging().getInitialNotification()` with the same FCM JSON, it's retuning:
```
RemoteMessage {
 data: {},
 notification: { body: 'new msg', title: 'Incoming Message' }
}
```

the data property is only empty object in iOS, with the same FCM JSON

**Reply by mikehardy** [reply]
> With the information you have provided, I can not say anything for certain.

https://stackoverflow.com/help/minimal-reproducible-example
Tools to create the above: https://github.com/mikehardy/rnfbdemo/blob/master/make-demo.sh
